tests/contrib/hooks/test_gcp_automl_hook.py

```diff
@@ -0,0 +1,275 @@
+# -*- coding: utf-8 -*-
+#
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+#
+import unittest
+
+from google.cloud.automl_v1beta1 import AutoMlClient, PredictionServiceClient
+
+from airflow import version
+from airflow.contrib.hooks.gcp_automl_hook import AutoMLHook
+from tests.compat import mock
+
+CREDENTIALS = "test-creds"
+TASK_ID = "test-automl-hook"
+GCP_PROJECT_ID = "test-project"
+GCP_LOCATION = "test-location"
+MODEL_NAME = "test_model"
+MODEL_ID = "projects/198907790164/locations/us-central1/models/TBL9195602771183665152"
+DATASET_ID = "TBL123456789"
+MODEL = {
+    "display_name": MODEL_NAME,
+    "dataset_id": DATASET_ID,
+    "tables_model_metadata": {"train_budget_milli_node_hours": 1000},
+}
+
+LOCATION_PATH = AutoMlClient.location_path(GCP_PROJECT_ID, GCP_LOCATION)
+MODEL_PATH = PredictionServiceClient.model_path(GCP_PROJECT_ID, GCP_LOCATION, MODEL_ID)
+DATASET_PATH = AutoMlClient.dataset_path(GCP_PROJECT_ID, GCP_LOCATION, DATASET_ID)
+
+INPUT_CONFIG = {"input": "value"}
+OUTPUT_CONFIG = {"output": "value"}
+PAYLOAD = {"test": "payload"}
+DATASET = {"dataset_id": "data"}
+MASK = {"field": "mask"}
+
+
+class TestAuoMLHook(unittest.TestCase):
+    def setUp(self) -> None:
+        self.hook = AutoMLHook()
```

Comment:
It's do a query to database. Mocking is necessary here. Please look at other operators.